djstripe/stripe_objects.py

```diff
@@ -84,22 +84,35 @@ class StripeObject(TimeStampedModel):
     class Meta:
         abstract = True
 
-    def api_retrieve(self, api_key=settings.STRIPE_SECRET_KEY):
+    @property
+    def default_api_key(self):
+        if self.livemode is None:
+            # Livemode is unknown. Use the default secret key.
+            return djstripe_settings.STRIPE_SECRET_KEY
+        elif self.livemode:
+            # Livemode is true, use the live secret key
+            return djstripe_settings.LIVE_API_KEY
+        else:
+            # Livemode is false, use the test secret key
+            return djstripe_settings.TEST_API_KEY
+
+    def api_retrieve(self, api_key=None):
         """
         Call the stripe API's retrieve operation for this model.
 
-        :param api_key: The api key to use for this request. Defaults to settings.STRIPE_SECRET_KEY.
+        :param api_key: The api key to use for this request. Defaults to djstripe_settings.STRIPE_SECRET_KEY.
```

Comment:
with these user-facing docstrings, it'd make sense to keep "settings.STRIPE_SECRET_KEY"